app/src/main/java/com/codesoom/assignment/DemoHttpHandler.java

```diff
@@ -0,0 +1,169 @@
+
+import com.codesoom.assignment.models.Task;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import java.io.*;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class DemoHttpHandler implements HttpHandler {
+
+    private List<Task> tasks = new ArrayList<>();
+
+    ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+
+        String method = exchange.getRequestMethod();
+        URI requestURI = exchange.getRequestURI();
+        String path = requestURI.getPath();
+        System.out.println(method + " " + path);
+
+        InputStream inputStream = exchange.getRequestBody();
+        String body = new BufferedReader(new InputStreamReader(inputStream))
+                .lines()
+                .collect(Collectors.joining("\n"));
+
+        // path가 아이디를 가지는지 여부
+        boolean hasTaskId = Pattern.matches("/tasks/[0-9]+$", path);
+
+        // 콘텐츠 초기화
+        String content = null;
+
+        // GET /tasks
+        if (method.equals("GET") && path.equals("/tasks")) {
+            content = tasksToJson(tasks);
+        }
+
+        // GET /tasks/{id}
+        if (method.equals("GET") && hasTaskId) {
+            Long id = extractIdFromPath(path);
+            Task findTask = findTaskById(id);
+            content = taskToJson(findTask);
+        }
+
+        // POST /tasks
+        if (method.equals("POST") && path.equals("/tasks")) {
+            Task task = createTaskWithId(body);
+            content = taskToJson(task);
+            tasks.add(task);
+        }
+
+        // Delete /tasks/{id}
+        if (method.equals("DELETE") && hasTaskId) {
+            Long id = extractIdFromPath(path);
+            Task findTask = findTaskById(id);
+            tasks.remove(findTask);
+        }
+
+        // PUT,PATCH /tasks/{id}
+        // 여기서는 Task 객체의 변경 가능한 필드가 title 뿐이므로 put, patch가 동일하게 동작한다.
+        if ((method.equals("PATCH") || method.equals("PUT")) && hasTaskId) {
+            Long id = extractIdFromPath(path);
+            Task findTask = findTaskById(id);
+            Task inputTask = toTask(body);
+            findTask.setTitle(inputTask.getTitle());
+        }
+
+        exchange.sendResponseHeaders(200, content.getBytes().length);
+        OutputStream responseBody = exchange.getResponseBody();
+        responseBody.write(content.getBytes());
+        responseBody.flush();
+        responseBody.close();
+    }
+
+    /**
+     * Task 객체 생성
+     * @param body
+     * @return
+     * @throws JsonProcessingException
+     */
+    private Task createTaskWithId(String body) throws JsonProcessingException {
```

Comment:
오오 javadoc을 작성하셨군요. 기왕 javadoc을 작성하신다면 잘 작성하는 것이 좋겠죠. 저를 따라서 다음과 같이 작성해 보세요.
```suggestion
    /**
     * http 요청의 body 문자열을 받아 할 일 객체를 생성해 리턴합니다.
     *
     * @param body http 요청의 body 문자열
     * @return Task 할 일
     * @throws JsonProcessingException json 처리중 문제가 발생한 경우
     */
    private Task createTaskWithId(String body) throws JsonProcessingException {
```

Reply:
감사합니다 종립님 열심히 따라해보겠습니다 ㅎㅎ 😁